src/stdlib/parse_regex_all.rs

```diff
@@ -105,22 +106,22 @@ impl Function for ParseRegexAll {
 #[derive(Debug, Clone)]
 pub(crate) struct ParseRegexAllFn {
     value: Box<dyn Expression>,
-    pattern: Regex,
+    pattern: Box<dyn Expression>,
     numeric_groups: Box<dyn Expression>,
 }
 
 impl FunctionExpression for ParseRegexAllFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
         let value = self.value.resolve(ctx)?;
         let numeric_groups = self.numeric_groups.resolve(ctx)?;
-        let pattern = &self.pattern;
+        let pattern = self.pattern.resolve(ctx)?.as_regex().unwrap().clone();
 
-        parse_regex_all(value, numeric_groups.try_boolean()?, pattern)
+        parse_regex_all(value, numeric_groups.try_boolean()?, &pattern)
     }
 
-    fn type_def(&self, _: &state::TypeState) -> TypeDef {
+    fn type_def(&self, _state: &state::TypeState) -> TypeDef {
         TypeDef::array(Collection::from_unknown(
-            Kind::object(util::regex_kind(&self.pattern)).or_null(),
+            Kind::object(Collection::any()).or_null(),
```

Comment:
`self.pattern.resolve_constant()` returns None

Reply:
> self.pattern.resolve_constant(..) returns None
We should make sure at least the case where the regex is passed in as a literal is supported (for backwards compatibility). In that case it should keep using `util::regex_kind`. Let me know if you need help with this, but I thought it would have just worked. It might just be a 1-liner to fix if it doesn't.

For the case where `resolve_constant` returns `None`, you can look at the `regex_kind` function to see what it inserts, and basically build a union of that. It seems like it only every creates an object where the values are bytes or null, but we won't know the key names since they can't be parsed in this situation. So you can create a `Kind::object` with a collection where the unknown type is `bytes | null`.

Reply:
Thanks, both cases make sense. Updated in https://github.com/vectordotdev/vrl/pull/563/commits/7ffc51f006e36650e057b919b34f8881897399ed. I will follow up with adding a test case (forgot to include in the latest commit).